tests/baselines/reference/assignmentToObjectAndFunction.errors.txt

```diff
@@ -14,6 +16,7 @@ tests/cases/compiler/assignmentToObjectAndFunction.ts(29,5): error TS2322: Type
 !!! error TS2322: Type '{ toString: number; }' is not assignable to type 'Object'.
 !!! error TS2322:   Types of property 'toString' are incompatible.
 !!! error TS2322:     Type 'number' is not assignable to type '() => string'.
+!!! error TS2322:       Type 'Number' provides no match for the signature '(): string'

```

Comment:
That's a good idea, but I looked around and didn't see an easy way to distinguish `globalNumberType`, for example, from any other type. Any ideas? I could add a function to check if `type === globalNumberType || type === ...`, but that seems like the wrong approach.